libcxx/include/__format/format_arg_store.h

```diff
@@ -188,15 +182,8 @@ _LIBCPP_HIDE_FROM_ABI basic_format_arg<_Context> __create_format_arg(_Tp& __valu
   else if constexpr (__arg == __arg_t::__unsigned_long_long)
     return basic_format_arg<_Context>{__arg, static_cast<unsigned long long>(__value)};
   else if constexpr (__arg == __arg_t::__string_view)
-    // Using std::size on a character array will add the NUL-terminator to the size.
-    if constexpr (is_array_v<_Dp>)
-      return basic_format_arg<_Context>{
-          __arg, basic_string_view<typename _Context::char_type>{__value, extent_v<_Dp> - 1}};
```

Comment:
It seems that we can still handle character arrays as `basic_string_view` provided `extent_v<_Dp> - 1` is replaced with the correct length. Is such method more preferred? @mordante

Reply:
If I understand correctly, this patch as-is will cause character arrays to decay to pointers, and then their size to be computed as-if `strlen` instead, right? Since we do have the information available statically, it might be interesting to carry it around to avoid recomputing this. However, I can also imagine this would require significant changes to the implementation.

Assuming I correctly understood what this tradeoff is about, one thing that might be worth checking is whether the `strlen` call is actually getting constant folded in most use cases. This may be a non-issue.

Reply:
Now I'm changing back to use `basic_string_view` to carry the length. It seems that we can diagnose UB in a straight forward way now.